### PR TITLE
feat(linter): eslint-plugin-unicorn prefer-reflect-apply(style)

### PR DESCRIPTION
**Pull request by Ken-HH24**
[prefer-reflect-apply](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-reflect-apply.md) for #684 .
I'm not pretty sure what the help message should be.

### CODE REVIEW

File: crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    /// foo.apply(null, [42]);
+    ///
+    /// // Good
+    /// Reflect.apply(foo, null);
+    /// ```
+    PreferReflectApply,
+    style
+);
+
+fn is_apply_signature(argument_1: &Argument, argument_2: &Argument) -> bool {
```
Comment (IWANABETHATGUY) [inline crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs:46]: NIT: 
```rs
fn is_apply_signature(argument_1: &Argument, argument_2: &Argument) -> bool {
    let (Argument::Expression(arg_expr_1), Argument::Expression(arg_expr_2)) =
        (argument_1, argument_2)
    else {
        return false;
    };

    if matches!(arg_expr_1, Expression::ThisExpression(_) | Expression::NullLiteral(_)) {
        if matches!(arg_expr_2, Expression::ArrayExpression(_)) {
            return true;
        }

        if let Expression::Identifier(iden) = arg_expr_2 {
            return iden.name == "arguments";
        }
    }

    false
}

```

File: crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+        if is_static_property_name_equal(member_expr, "apply")
+            && call_expr.arguments.len() == 2
+            && is_apply_signature(&call_expr.arguments[0], &call_expr.arguments[1])
+        {
+            ctx.diagnostic(PreferReflectApplyDiagnostic(call_expr.span));
+            return;
+        }
+
+        if is_static_property_name_equal(member_expr, "call") {
```
Comment (IWANABETHATGUY) [inline crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs:100]: NIT: 
```rs
        if is_static_property_name_equal(member_expr, "call") {
            let Expression::MemberExpression(member_expr_obj) = member_expr.object() else {
                return;
            };
            if is_static_property_name_equal(member_expr_obj, "apply") {
                let Expression::MemberExpression(member_expr_obj_obj) = member_expr_obj.object()
                else {
                    return;
                };
                if is_static_property_name_equal(member_expr_obj_obj, "prototype") {
                    let Expression::Identifier(iden) = member_expr_obj_obj.object() else { return };
                    if iden.name == "Function"
                        && matches!(call_expr.arguments.as_slice(), [_, second, third] if is_apply_signature(&second, &third))
                    {
                        ctx.diagnostic(PreferReflectApplyDiagnostic(call_expr.span));
                    }
                }
            }
        }

```

File: crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs (hunk 1 above)
Comment (IWANABETHATGUY) [reply crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs:46]: or just 
```rs
fn is_apply_signature(argument_1: &Argument, argument_2: &Argument) -> bool {
    match argument_1 {
        Argument::Expression(Expression::ThisExpression(_) | Expression::NullLiteral(_)) => {
            matches!(argument_2, Argument::Expression(Expression::ArrayExpression(_)))
                || matches!(argument_2, Argument::Expression(Expression::Identifier(ident)) if ident.name == "arguments")
        }
        _ => false,
    }
}

```

File: crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs (hunk 2 above)
Comment (Ken-HH24) [reply crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs:86]: Thanks for your advice. I've made some adjustments.

File: crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs (hunk 1 above)
Comment (Ken-HH24) [reply crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs:46]: Thanks for your advice.

**Review by Boshen — APPROVED** [review]

**Pull request merged**